internal/controller/apiextensions/composite/composition_transforms.go

```diff
@@ -281,7 +281,7 @@ func unmarshalJSON(j extv1.JSON, output *any) error {
 }
 
 // ResolveString resolves a String transform.
-func ResolveString(t v1.StringTransform, input any) (string, error) {
```

Comment:
otherwise you should update this to also return nil, nil for string transforms https://github.com/crossplane/crossplane/blob/296ee5d5dfdfb70bd23a217c7ab29448d0580933/apis/apiextensions/v1/composition_transforms.go#L127

Reply:
That was something bugging me as well, but felt okay after framing it as "string converts would convert either from or to a string"

Currently, ToJson already takes an object as input and produces a string.
With this, FromJson will take a string as input and produce an object.

How does this sound?

Reply:
> would it make sense to make this a ConvertTransfrom from string to object?

Do you have any idea on how it would look like in the API?

Reply:
here the pr with the fix: https://github.com/crossplane/crossplane/pull/4527.

However, yes I'd rather go with a convert, as accepting multiple inputs type is fine, as we can still proceed with the chain of validation, while producing an arbitrary type as output will just make impossible validating types after that transform as we won't know what type to assume.

Reply:
Thanks!

Not sure I got the difference 100% regarding the validation, but I still prefer to continue with a string convert transform due to the following:
- We already have `ToJson` there, and having `FromJson` as a complement would make it more discoverable.
- Compared to a `type: convert` with `toType: object`, it is more clear/explicit that we expect a Json string as input.
- I have some concerns around `toType: object`, as provided Json could represent a map/list or whatever and could be confusing.

Reply:
Ok, fair enough. 

Just updated accordingly 👍